### DISCUSSION TITLE
How to call a react component from html

### DISCUSSION THREAD

**Discussion by PerfekDev** [community/community#58887] [Programming Help] [answered]
### Body

I have define a react component that receive parameters

```
import React from 'react';
import ReactDOM from 'react-dom/client';
import DocumentEditor from './DocumentEditor';

export default function DocumentEditorContainer(props){
    //let props = {container : "root", value: ""};
    const root = ReactDOM.createRoot(document.getElementById(props.container));
    root.render(
      <React.StrictMode>
        <DocumentEditor value = {props.value}/>
      </React.StrictMode>
    );
}
```

I package it using webpack, and I need to call that module from a script tag inside a html file

```
<script type="text/javascript" src="./aux.js"></script>
<script>
        DocumentEditorContainer("root", "123");
</script>
```

But I am getting that DocumentEditorContainer is undefined. What am I doing wrong?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by leonden** [comment]
The issue you're facing is that you're not importing the `DocumentEditorContainer` component correctly in your HTML file. Here's how you can fix it:

1. Make sure you have bundled your React component file (aux.js) using webpack or a similar tool.

2. In your HTML file, include the bundled JavaScript file generated by webpack, like this:
```html
<script type="text/javascript" src="path/to/bundle.js"></script>
```
Replace `"path/to/bundle.js"` with the actual path to your bundled JavaScript file.

3. After including the JavaScript file, you can call the `DocumentEditorContainer` function. However, there's a small mistake in the way you're passing the arguments. Update your script tag as follows:
```html
<script>
  DocumentEditorContainer({ container: "root", value: "123" });
</script>
```

By passing an object with properties `container` and `value` as an argument, you can correctly provide the required parameters to the `DocumentEditorContainer` component.

With these changes, the `DocumentEditorContainer` function should be recognized and executed properly.

**Comment by iamsuiux** [comment, marked as answer]
From the code snippets you've shared, it seems like you're trying to use the DocumentEditorContainer function from an external script in your HTML file. However, the problem is that this function isn't globally accessible because it's bundled with webpack.

When webpack bundles your code, it encapsulates your modules in closures, and thus they aren't added to the global scope. As a result, when you try to call DocumentEditorContainer from your script tag, it's undefined because it's not in the global scope.

To solve this issue, you need to expose your DocumentEditorContainer to the global scope. I've never used it in a project, but I'll give it a try, as I think it should work.

Basically, you can do this by attaching the function to the window object in your JavaScript file.
